File: zezar/zezar.py
# Solves "donKEY Kong" from the 2017 HSPC at UVa
# Solution by Joseph Tobin, 2017

from sys import stdin

if __name__ == '__main__':
  data = stdin.read().splitlines()
  n = int(data[0])
  i = 1
  for case in range(n):
    c_solved = data[i]
    i = i+1;
    p_solved = data[i];
    i = i + 1
    c_unsolved = data[i]
    i = i + 1

    ans = ""

    key = (ord(c_solved[0]) - ord(p_solved[0])) % 26

    for c in c_unsolved:
      if(not (c == "_")) :
        ans += chr( (((ord(c)%65) - key) % 26) + 65 )
      else :
        ans += "_"

    print("Case " + str(case + 1) + ": " + ans)

File: zezar/zezar.cpp
// Solves "donKEY Kong" from the 2017 HSPC at UVa
// Solution by Aaron Bloomfield, 2017

#include <iostream>
#include <string>
using namespace std;
int main() {
  string a, b;
  unsigned int key, n;
  cin >> n;
  for ( unsigned int i = 0; i < n; i++ ) {
    cin >> a >> b;
    key = a[0]-b[0];
    if ( key >= 26 )
      key -= 26;
    if ( key < 0 )
      key += 26;
    cin >> a;
    cout << "Case " << (i+1) << ": ";
    for ( unsigned int j = 0; j < a.length(); j++ ) {
      char c = a[j];
      if ( (c >= 'A') && (c <= 'Z') ) {
	c -= key;
	if ( c < 'A' )
	  c += 26;
      }
      cout << c;
    }
    cout << endl;
  }
}
